biblib/views/user_view.py

```diff
@@ -153,16 +160,13 @@ def get_libraries(cls, service_uid, absolute_uid, start=0, rows=None, sort_col="
                     owner=owner
                 )
 
-                if (ownership and main_permission in ['owner']) or not ownership: 
-                    my_libraries.append(payload)
-                elif ownership and main_permission in ['admin', 'read', 'write']: 
-                    shared_with_me.append(payload)
-            
-            if ownership: 
-                libraries_response['my_libraries'] = my_libraries 
-                libraries_response['shared_with_me'] = shared_with_me
-            else: 
-                libraries_response['libraries'] = my_libraries
+                if type == 'all' or (type == 'owner' and main_permission in ['owner']) or (type == 'collaborator' and main_permission in ['admin', 'read', 'write']): 
```

Comment:
Maybe I am missing something, but do we still need this `if` with the filtering incorporated in the new method you wrote above?

Reply:
Yes, seems like this part and the slicing of list (below) are not needed anymore